users/users-model.js

```diff
@@ -0,0 +1,32 @@
+const knex = require('knex');
+
+const dbConfig = require('../knexfile.js');
+
+const db = knex(dbConfig.development);
+
+module.exports = {
+  add,
+  find,
+  findBy,
+  findById,
+};
+
+function find() {
+  return db('users').select('id', 'username', 'password');
+}
+
+function findBy(filter) {
+  return db('users').where(filter);
+}
+
+async function add(user) {
+  const [id] = await db('users').insert(user);
+
+  return findById(id);
+}
+
+function findById(id) {
+  return db('users')
+    .where({ id })
```

Comment:
Why not use your `findBy` helper here? For example:
```javascript
  return findBy({ id }).first();
```

Reply:
That actually makes a lot of sense.  Making that change now